core/src/main/java/org/elasticsearch/action/bulk/BulkShardRequest.java

```diff
@@ -89,15 +82,14 @@ public void readFrom(StreamInput in) throws IOException {
                 items[i] = BulkItemRequest.readBulkItem(in);
             }
         }
-        refresh = in.readBoolean();
     }
 
     @Override
     public String toString() {
         // This is included in error messages so we'll try to make it somewhat user friendly.
         StringBuilder b = new StringBuilder("BulkShardRequest to [");
         b.append(index).append("] containing [").append(items.length).append("] requests");
-        if (refresh) {
+        if (refresh()) {
```

Comment:
this is strange in a `toString()` method. It seems like you are actually mutating the value, can you make this `isRefresh()`